### PR TITLE
Test for creating tags for files

### PR DESCRIPTION
**Pull request by skshetry**
## Related Issue

- Related to #32843 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/WebUIFilesContext.php [hunk 1]
```diff
@@ -210,7 +212,7 @@ public function theUserBrowsesToTheFilesPage() {
 	 * @throws Exception
 	 */
 	public function theUserBrowsesDirectlyToDetailsTabOfFileInFolder(
-		$tabName, $fileName, $folderName
+		$tabName=null, $fileName, $folderName
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUIFilesContext.php]: we usually put spaces like:
```
$tabName = null
```
also it is not really useful to have an optional parameter in the middle of the list.

Let's see what happens when the full test suite runs, maybe Behat knows how to call this method in the different ways.

**Review by phil-davis — COMMENTED** [review]
It should work. Just put the spaces around the ``=``

File: tests/acceptance/features/bootstrap/WebUIFilesContext.php (hunk 1 above)
Comment (skshetry) [reply tests/acceptance/features/bootstrap/WebUIFilesContext.php]: @phil-davis I fixed that. Please review again

**Comment by skshetry** [comment]
Backport for `stable10` on #32884

File: tests/acceptance/features/webUITags/createTags.feature [hunk 2]
```diff
@@ -0,0 +1,28 @@
+@webUI @insulated @systemtags-app-required
+Feature: Creation of tags for the files and folders
+  As a user
+  I want to create tags for the files/folders
+  So that I can find them easily
+
+  Background:
+    Given these users have been created:
+      | username | password | displayname | email        |
+      | user1    | qwerty   | User One    | u1@oc.com.np |
```
Comment (phil-davis) [inline tests/acceptance/features/webUITags/createTags.feature]: Use the regular user password by putting:
```
%regular%
```
in the password column, and in the step 2 lines below.

**Review by phil-davis — COMMENTED** [review]
We need to use the password substitutions nowadays, so that different passwords can be automatically passed in if needed.

**Review by phil-davis — APPROVED** [review]

File: tests/acceptance/features/webUITags/createTags.feature [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  As a user
+  I want to create tags for the files/folders
+  So that I can find them easily
+
+  Background:
+    Given these users have been created:
+      | username | password  | displayname | email        |
+      | user1    | %regular% | User One    | u1@oc.com.np |
+    And the user has browsed to the login page
+    And the user has logged in with username "user1" and password "qwerty" using the webUI
```
Comment (phil-davis) [inline tests/acceptance/features/webUITags/createTags.feature]: I expect that this login will fail.
The password neeeds to be ``%regular%``

**Review by phil-davis — CHANGES_REQUESTED** [review]
This needs matrix entries added in ``.drone.yml``
Currently drone is not actually running the tests in the new suite. That is why the test run is passing.

Also, please rebase to current master. master was updated to have extra PHP code style checks, so we want to be sure that those pass.

**Comment by skshetry** [comment]
@phil-davis  I fixed that password and added config in `drone.yml`. I ran the new acceptance-test-style too

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
@ownclouders rebase

**Comment by ownclouders** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by ownclouders** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

**Comment by skshetry** [comment]
I should also to wait for the dropdown to appear. Locally, it is very fast and doesn't need to wait. But, it seems that it's slower on the ci.

**Comment by phil-davis** [comment]
OK, I will not merge this today. We can look again on Monday.

**Comment by skshetry** [comment]
@phil-davis now it will wait for the dropdown and should pass.

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
@skshetry one of the new tags scenarios failed:
https://drone.owncloud.com/owncloud/core/10990/515

But in stable10 it passed.

???

**Comment by skshetry** [comment]
@phil-davis I have rebased with master. Maybe it's not this, but I don't know. I am out of options. Let's see if it passes.

File: tests/acceptance/features/webUITags/createTags.feature [hunk 4]
```diff
@@ -0,0 +1,28 @@
+@webUI @insulated @systemtags-app-required
+Feature: Creation of tags for the files and folders
+  As a user
+  I want to create tags for the files/folders
+  So that I can find them easily
+
+  Background:
+    Given these users have been created:
+      | username | password  | displayname | email        |
```
Comment (phil-davis) [inline tests/acceptance/features/webUITags/createTags.feature]: @skshetry Please adjust this table and rebase. The way users are created changed at the weekend.

Comment (skshetry) [reply tests/acceptance/features/webUITags/createTags.feature]: Fixed.

**Review by phil-davis — APPROVED** [review]

**Pull request merged**